disnake/interactions/message.py

```diff
@@ -82,7 +85,7 @@ def __init__(self, *, data: MessageInteractionPayload, state: ConnectionState) -
         self.data: MessageInteractionData = MessageInteractionData(
             data=data["data"], state=state, guild_id=self.guild_id
         )
-        self.message = Message(state=self._state, channel=self.channel, data=data["message"])
+        self.message = Message(state=self._state, channel=self.channel, data=data["message"])  # type: ignore # pyright bug
```

Comment:
what a funky bug :thinking: especially since the language server can clearly resolve it when hovering over it

Looks like weird descriptor behavior, since replacing the `cached_slot_property` for `.channel` with a standard `property` fixes the problem. I might try to reproduce it later and file an issue, if you haven't done so already

Reply:
forgot to update this; the correct way to do this is using `Type[Any]` in the descriptor instead:
```py
+++ b/disnake/utils.py
@@ -141,14 +141,14 @@ class CachedSlotProperty(Generic[T, T_co]):
         self.__doc__ = function.__doc__
 
     @overload
-    def __get__(self, instance: None, owner: Type[T]) -> Self:
+    def __get__(self, instance: None, owner: Type[Any]) -> Self:
         ...
 
     @overload
-    def __get__(self, instance: T, owner: Type[T]) -> T_co:
+    def __get__(self, instance: T, owner: Type[Any]) -> T_co:
         ...
 
-    def __get__(self, instance: Optional[T], owner: Type[T]) -> Any:
+    def __get__(self, instance: Optional[T], owner: Type[Any]) -> Any:
         if instance is None:
             return self
```

This gets rid of the error with `@cached_slot_property` in generic subclasses.